src/core/manager.c

```diff
@@ -3288,6 +3303,9 @@ int manager_reload(Manager *m) {
         if (q < 0 && r >= 0)
                 r = q;
 
+        if (!MANAGER_IS_RELOADING(m))
```

Comment:
It's quite possible I'm overthinking this, but: reloading state in manager uses a counter (n_reloading), not just a bool, so reload during reload is at least potentially possible. And this should be run only when n_reloading == 0, otherwise the signal(s) will be not sent.

This is one of the reasons why the saving of jobs for later use is done at all. (The other one is I wasn't sure it's right to use dbus before manager_recheck_dbus() is done.)